### PR TITLE
Fixed documentation link

### PR DESCRIPTION
**Pull request by sandeshkaranam**
**Overview**
Documentation links were broken due to missing file extensions

**Screenshots (if applicable)**

**Documentation**
- [x] Updated Documentation links

### CODE REVIEW

File: docs/documentation/components/portal.mdx [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 The Portal component is a utility component to help with the `ReactDOM.createPortal` API.
 Learn more about portal on the [official React docs.](https://reactjs.org/docs/portals.html) <br />
-Portals are used throughout Evergreen in components such as [Overlay](./overlay) and [Positioner](./positioner) to provide
+Portals are used throughout Evergreen in components such as [Overlay](./overlay.mdx) and [Positioner](./positioner.mdx) to provide
```
Comment (mshwery) [inline docs/documentation/components/portal.mdx:5]: Positioner link works fine as-is, but Overlay results in 404.

**Review by mshwery — COMMENTED** [review]
I'm not able to repo broken links on anything except Overlay, because that's not a component that exists anymore.

File: docs/documentation/components/portal.mdx (hunk 1 above)
Comment (kamebkj) [reply docs/documentation/components/portal.mdx:5]: `Overlay` is still a component in the v6 package but we missed it on our doc site. Will be opening a PR to address this.

**Comment by kamebkj** [comment]
Fixed the `Overlay` one with https://github.com/segmentio/evergreen/pull/1336. Cannot repo the other links.  
Thanks for the good catch!

**Pull request closed without merging**